smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/GradleGenerator.kt

```diff
@@ -9,78 +9,202 @@ import software.amazon.smithy.kotlin.codegen.KotlinSettings
+    dependencies: List<KotlinDependency>,
+    pluginsRenderer: InlineCodeWriter,
+    repositoryRenderer: InlineCodeWriter,
+    annotationRenderer: InlineCodeWriter
+) {
+    writer.write(
+        """
+        plugins {
+            #W
+        }  
+        #W
+        
+        kotlin {
+            jvm {
+                compilations.all {
+                    kotlinOptions.jvmTarget = "#L"
+                }
+                testRuns["test"].executionTask.configure {
+                    useJUnit()
+                }
+            }
+            sourceSets {
+                val commonMain by getting {
+                    dependencies {
+                        #W
+                    }
+                }
+                val jvmMain by getting
+            }
+            val optInAnnotations = listOf(
+                #W
+            )
+            kotlin.sourceSets.all {
+                optInAnnotations.forEach { languageSettings.optIn(it) }
+            }
+        }
+        """.trimIndent(),
+        pluginsRenderer,
+        { w: CodeWriter -> if (isRootModule) repositoryRenderer(w) },
+        JVM_TARGET_VERSION,
+        { w: CodeWriter -> renderDependencies(w, isSrcScope = true, isKmp = true, dependencies = dependencies) },
+        annotationRenderer
+    )
+}
+
+fun renderJvmGradleBuild(
+    writer: CodeWriter,
+    isRootModule: Boolean,
+    dependencies: List<KotlinDependency>,
+    pluginsRenderer: InlineCodeWriter,
+    repositoryRenderer: InlineCodeWriter,
+    annotationRenderer: InlineCodeWriter
+) {
+    writer.write(
+        """
+        plugins {
+            #W
+        }
+        #W
+
+        dependencies {
+            #W
+        }
+        val optinAnnotations = listOf(
+            #W
+        )
+        kotlin.sourceSets.all {
+            optinAnnotations.forEach { languageSettings.optIn(it) }
+        }
+
+        tasks.test {
+            useJUnitPlatform()
+            testLogging {
+                events("passed", "skipped", "failed")
+                showStandardStreams = true
+            }
+        }
+        """.trimIndent(),
+        pluginsRenderer,
+        { w: CodeWriter -> if (isRootModule) repositoryRenderer(w) },
+        { w: CodeWriter -> renderDependencies(w, isSrcScope = true, isKmp = false, dependencies = dependencies) },
+        annotationRenderer
+    )
+}
+
+private fun renderDependencies(writer: CodeWriter, isSrcScope: Boolean, isKmp: Boolean, dependencies: List<KotlinDependency>) {
+    if (!isKmp) {
+        writer.write("implementation(kotlin(\"stdlib\"))")
+    }
+
+    // TODO - can we make kotlin dependencies not specify a version e.g. kotlin("kotlin-test")
+    // TODO - Kotlin MPP setup (pass through KotlinSettings) - maybe separate gradle writers
+    val orderedDependencies = dependencies.sortedWith(compareBy({ it.config }, { it.artifact }))
+    orderedDependencies
+        .filter {
+            if (isSrcScope) !it.config.isTestScope else it.config.isTestScope
```

Comment:
**Nit**: Boolean logic can be simplified:

`.filter { it.config.isTestScope != isSrcScope }`

This highlights the incongruity in these flags. We have one local flag for source scope and one broader flag for test scope. We might be able to clean this up by making the local flag refer to test scope or by changing both to an enum or anything else that unifies their intent.